### DISCUSSION TITLE
tud_cdc_connected() or tud_cdc_line_state_cb() problem when unplug USB connector

### DISCUSSION THREAD

**Discussion by mastupristi** [hathach/tinyusb#1185] [Q&A]
hi,
I'm using tinyUsb 0.12.0, on i.MX RT1051.
I'm successfully using the `tud_cdc_line_state_cb()` callback to be notified when on the PC I connect/disconnect to the ttyACM0 (on PC I use `minicom`).
The scenario is: with the usb cable plugged in I run minicom and the `tud_cdc_line_state_cb()` is called with `dtr=1`. In the moment I quit minicom the `tud_cdc_line_state_cb()` is called with `dtr=0`.
The problem arises if, with minicom running, i unplug the usb cable. In that case the `tud_cdc_line_state_cb()` is not called. But when I plug the cable back in it is called again with `dtr=1`.
This makes it difficult if not impossible to code state machines take into account the connection state.

Is this behavior intentional? How can I be notified of the cable disconnection?

best regards
Max

**Comment by hathach** [comment]
- you could use tud_umount_cb() which is invoked when usb cable is disconnected. 
- dtr is set again because minicom try to re-connect afterwards

**Reply by mastupristi** [reply]
I just tried.
Unfortunately, `tud_umount_cb()` is not called either when I unplug the cable.

best regards
Max

**Reply by hathach** [reply]
can you try out the stock cdc_msc example, then
1. power device without plugging it it: Led should blink fast
2. plug usb cable: led should blink normal
3. unplug usb cable: check whether it blink fast or normal ?

**Reply by mastupristi** [reply]
I tried stock cdc_msc_freertos example on IMXRT1050-EVKB:

1. I power device without plugging it it: Led blink fast
2. I plug usb cable: led blink normal
3. I unplug usb cable: led still blink normal

tried also stock cdc_msc still on on IMXRT1050-EVKB: same behaviour

Neither detects cable disconnection

best regards
Max

**Reply by hathach** [reply]
Thank you for your confirmation, it is then a bug on imrxt controller driver. I will create an issue based on this discussion
- #1187
Will try to test/fix it later on.
